cart/application/cartService.go

```diff
@@ -62,8 +62,24 @@ type (
 
 	// PromotionFunction type takes ctx, cart, couponCode and applies the promotion
 	promotionFunc func(context.Context, *cartDomain.Cart, string) (*cartDomain.Cart, cartDomain.DeferEvents, error)
+
+	contextKeyType string
+)
+
+const (
+	itemIDKey contextKeyType = "item_id"
 )
 
+func ItemIDFromContext(ctx context.Context) string {
+	id, _ := ctx.Value(itemIDKey).(string)
+
+	return id
+}
+
+func contextWithItemID(ctx context.Context, itemID string) context.Context {
```

Comment:
So we don't plan on making this public? So when i want to call the restrictor from places else then cart service?